Add test coverage

rclee opened the pull request.

rclee committed:
Add test coverage to MMEEventsManagerTests

rclee committed:
Test metadata POST

rclee committed:
Update MMEAPIClientTests.m

rclee committed:
Add clregion distance filter test

rclee committed:
Update MetricsManager tests

rclee committed:
Update MMEMetricsManagerTests

Removes obsolete conditionals in MMEMetricsManager

rclee committed:
Port MMEAPIClientTests and test unzipped data

rclee committed:
Create MMEEventTests.m

rclee committed:
Comment unused methods and add TODO

rclee committed:
Update MMEConfigurationTests

rclee committed:
Port MMEUniqueIdentifierTests to XCTest

Remove init method (not needed with the unavailable attribute in the header)

rclee committed:
Remove logDateFormatter and add MMEDate XCTest

rclee commented:
Majority of this is done if you have some time to look it over @alfwatt 👀

rclee committed:
Add to MMEDateTests

rclee commented:
`MMECertPin` and `MMEDate` has a few more things that could be tested. After we remove the debug classes (that don't need tests here) we should be over 90% on code coverage.

alfwatt commented on Tests/NSUserDefaults+MMEConfigurationTests.m:
```diff
@@ -87,6 +87,77 @@ - (void)testConfigEventTagDefault {
+    XCTAssert([NSUserDefaults.mme_configuration.mme_legacyHostSDKVersion isEqualToString:@"1.0.0"]);
+}
+
+//TODO: Persistent object deletion isn't working as expected
```
We should resolve this before merging these changes

alfwatt replied:
@rclee looking at testDeletePersistentObject we probably shoudn’t be mixing that test with MMECollectionDisabled which has some special cases

I suggest re-writing the test to use mme_setObject:forPersistentKey: (which you’ll need to add to the private header, maybe along with it’s volatile counterpart

alfwatt commented on MapboxMobileEvents/Categories/NSString+MMEVersions.m:
```diff
@@ -106,44 +106,45 @@ - (NSInteger)mme_semverPatchVersion {
-- (NSArray<NSString *>*)mme_semverPreReleaseIdentifiers {
-    return [[self mme_semverPreReleaseComponent] componentsSeparatedByString:@"."];
-}
+//TODO: These methods are not working as expected in tests and are not used currently
```
IIRC there aren't tests for these, we should add some if we're going to keep the code at all (given it's not currently used, it could be deleted from the project until it's needed)

alfwatt reviewed (commented):
Couple of smaller issues

rclee committed:
Remove unused code

When writing a couple of the tests it wasn't behaving as I would have expected. Removing the unused code for now seems like a good bet.

rclee committed:
Add Persistent and Volatile object deletion tests

alfwatt reviewed (approved):
LGTM

rclee committed:
Merge branch 'master' into rclee-test-100

rclee committed:
Merge conflict error
